core/dom/rect.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @fileOverview Defines the "virtual" {@link CKEDITOR.dom.rect} class
+ * which contains the defintion of an element's DOM rectangle. This file is for
+ * documentation purposes only.
+ */
+
+/**
+ * Virtual class that illustrates the {@link CKEDITOR.dom.element} DOM rectangle (dimensions and coordinates
+ * of the area, which element occupies on the page).
+ *
+ * @class CKEDITOR.dom.rect
+ * @abstract
+ * @since 4.10.0
+ */
+
+/**
+ * Element's offset from the top.
+ *
+ * @property {Number} top
+ */
+
+/**
+ * Element's offset from the bottom.
```

Comment:
By offset from bottom I would understand distance to the bottom edge of the viewport. This is not the case.

Bottom contains offset from the top edge of the viewport.